src/pages/ProposalDetailPage.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { OidcSecure } from '@axa-fr/react-oidc';
+import { useParams } from 'react-router-dom';
+import { usePdcApi } from '../use-pdc-api';
+import { ProposalDetailPanel } from '../components/ProposalDetailPanel';
+
+interface CanonicalField {
+  id: number;
+  label: string;
+  shortCode: string;
+}
+
+interface Proposal {
+  versions: {
+    version: number;
+    fieldValues: {
+      applicationFormField: {
+        canonicalFieldId: number;
+        position: number;
+      };
+      value: string;
+    }[];
+  }[];
+}
+
+const getValueOfCanonicalField = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+  canonicalFieldShortCode: string,
+) => {
+  const field = canonicalFields.find(({ shortCode }) => (
+    shortCode === canonicalFieldShortCode
+  ));
+  if (field === undefined) {
+    return undefined;
+  }
+  const fieldValue = proposal.versions[0].fieldValues.find(({ applicationFormField }) => (
+    applicationFormField.canonicalFieldId === field.id
+  ));
+  return fieldValue?.value ?? undefined;
+};
+
+const mapCanonicalFields = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+) => (
+  proposal.versions[0].fieldValues.map(({ applicationFormField, value }) => ({
+    fieldName: canonicalFields.find(({ id }) => (
+      id === applicationFormField.canonicalFieldId
+    ))?.label ?? 'missing',
+    position: applicationFormField.position,
+    value,
+  }))
+);
+
+const ProposalDetailPage = () => {
+  const params = useParams();
+  const proposalId = params.proposalId ?? 'missing';
+  const canonicalFields = usePdcApi<CanonicalField[]>('/canonicalFields');
+  const proposal = usePdcApi<Proposal>(`/proposals/${proposalId}?includeFieldsAndValues=true`);
+
+  if (canonicalFields === null || proposal === null) {
+    return (
+      <OidcSecure>
+        <ProposalDetailPanel
+          title="Loading..."
+          author="Loading..."
+          authorId="00-0000000"
+          version={0}
+          values={[]}
+        />
+      </OidcSecure>
+    );
+  }
+
+  const title = getValueOfCanonicalField(canonicalFields, proposal, 'proposal_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_title')
+    ?? `Proposal ${proposalId}`;
+  const author = getValueOfCanonicalField(canonicalFields, proposal, 'organization_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'organization_dba_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'organization_legal_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_primary_contact_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_submitter_name')
+    ?? 'Unknown Author';
+  const authorId = getValueOfCanonicalField(canonicalFields, proposal, 'organization_tax_id');
```

Comment:
@jim-mcgowan I'd appreciate your thoughts on this priority list for picking canonical fields to fill the header! In case the code is unclear, it picks the first present value, so for title it'd look for `proposal_name`, then if that's absent `proposal_title`, then if that's absent it'll construct a name like `Proposal 7`.

Reply:
I talked with @jim-mcgowan on the phone, and my takeaway was that this is good for now, so resolving.